File: Plants.Core/Services/CategoryService.cs
using Plants.Core.Entities;
using Plants.Core.IRepositories;
using Plants.Core.IServices;
using System;
using System.Collections.Generic;
using System.Text;

namespace Plants.Core.Services
{
    public class CategoryService : ICategoryService
    {
        private ICategoryRepository<Category> _categoryRepository;

        public CategoryService(ICategoryRepository<Category> categoryRepository)
        {
            _categoryRepository = categoryRepository;
        }

        public Category Add(Category category)
        {
            return _categoryRepository.Add(category);
        }

        public ICollection<Category> GetAll()
        {
            return _categoryRepository.GetAll();
        }

        public Category GetByID(Guid? ID)
        {
            return _categoryRepository.GetByID(ID);
        }

        public Category Update(Category category)
        {
            return _categoryRepository.Update(category);
        }

        public void Delete(Guid? ID)
        {
            _categoryRepository.Delete(ID);
        }
    }
}

File: Plants.API/ClientApp/src/app/services/category.service.ts
import { Injectable } from "@angular/core";
import { HttpClient } from "@angular/common/http";
import { CategoryModel } from "../models/category.model";
import { Guid } from "guid-typescript";
import { Observable } from "rxjs";

@Injectable({ providedIn: 'root' })
export class CategoryService {
    private url = "";

    constructor(private http: HttpClient) {}

    public add(model: CategoryModel) : Observable<CategoryModel> {
        return this.http.post<CategoryModel>(this.url + "category/add", model);
    }

    public getAll() : Observable<CategoryModel[]> {
        return this.http.get<CategoryModel[]>(this.url + "category/all");
    }

    public getByID(id: Guid) : Observable<CategoryModel> {
        return this.http.get<CategoryModel>(this.url + "category/" + id);
    }

    public update(model: CategoryModel) : Observable<CategoryModel>{
        return this.http.put<CategoryModel>(this.url + "category/update", model)
    }

    public delete(id: Guid) {
        return this.http.delete(this.url + "category/" + id)
    }
}